lib/iris/util.py

```diff
@@ -1080,6 +1080,9 @@ def new_axis(src_cube, scalar_coord=None):
     if scalar_coord is not None:
         scalar_coord = src_cube.coord(scalar_coord)
 
+    if isinstance(src_cube.data, np.ma.core.MaskedConstant):
```

Comment:
Using `src_cube.data` makes this function load deferred data, which is why you saw the deferred data test fail. Can you formulate this check without loading the data?